extension/src/experiments/columns/model.ts

```diff
@@ -1,17 +1,22 @@
-import { Memento } from 'vscode'
+import { EventEmitter, Memento } from 'vscode'
 import { collectChanges, collectColumns, collectParamsFiles } from './collect'
 import { Column, ColumnType } from '../webview/contract'
 import { ExperimentsOutput } from '../../cli/dvc/contract'
 import { PersistenceKey } from '../../persistence/constants'
 import { PathSelectionModel } from '../../path/selection/model'
 
 export class ColumnsModel extends PathSelectionModel<Column> {
+  private columnsOrderChanged: EventEmitter<void>
   private columnOrderState: string[] = []
   private columnWidthsState: Record<string, number> = {}
   private columnsChanges: string[] = []
   private paramsFiles = new Set<string>()
 
-  constructor(dvcRoot: string, workspaceState: Memento) {
+  constructor(
+    dvcRoot: string,
+    workspaceState: Memento,
+    columnsOrderChanged: EventEmitter<void>
```

Comment:
Tooltips are only built once unless you request a rebuild. To make sure that the tooltips had the latest first three columns, I added an `EventEmitter` to the `ColumnsModel`.